### PR TITLE
Building helm chart v3 migration tool binaries

### PR DESCRIPTION
**Pull request by rnishtala-sumo**
Building helm chart v3 migration tool binaries

Binaries from a test run can be found below:
https://github.com/SumoLogic/sumologic-kubernetes-tools/actions/runs/3941261582

### CODE REVIEW

File: src/go/cmd/update-collection-v3/Makefile [hunk 1]
```diff
@@ -1,5 +1,10 @@
 .PHONY: build
 build:
-	go build .
+	env GOOS=windows GOARCH=arm64 go build -o update-collection-v3-sumo-windows_arm64
+	env GOOS=windows GOARCH=amd64 go build -o update-collection-v3-sumo-windows_amd64
+	env GOOS=linux GOARCH=arm64 go build -o update-collection-v3-sumo-linux_arm64
+	env GOOS=linux GOARCH=amd64 go build -o update-collection-v3-sumo-linux_amd64
+	env GOOS=darwin GOARCH=arm64 go build -o update-collection-v3-sumo-darwin_arm64
+	env GOOS=darwin GOARCH=amd64 go build -o update-collection-v3-sumo-darwin_amd64
```
Comment (sumo-drosiek) [inline src/go/cmd/update-collection-v3/Makefile:8]: I would go with iteration

File: .github/workflows/build_binaries.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        with:
+          name: update-collection-v3-sumo-darwin_amd64
+          path: ./src/go/cmd/update-collection-v3/update-collection-v3-sumo-darwin_amd64
+          if-no-files-found: error
+      - name: Store darwin (arm64) binary as action artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: update-collection-v3-sumo-darwin_arm64
+          path: ./src/go/cmd/update-collection-v3/update-collection-v3-sumo-darwin_arm64
+          if-no-files-found: error
```
Comment (sumo-drosiek) [inline .github/workflows/build_binaries.yml:52]: I would go with wildcard pattern: https://github.com/actions/upload-artifact#upload-using-a-wildcard-pattern

**Review by sumo-drosiek — APPROVED** [review]
LGTM functionally

File: .github/workflows/build_binaries.yml (hunk 2 above)
Comment (rnishtala-sumo) [reply .github/workflows/build_binaries.yml:52]: it seems to upload one zip with all the binaries file with the wildcard pattern, do we want that?

Comment (sumo-drosiek) [reply .github/workflows/build_binaries.yml:52]: in that case we don't :)

**Review by swiatekm-sumo — APPROVED** [review]
I was going to say to update release.md, but it looks like we don't have it in this repo. Should be added in a separate PR.

File: src/go/cmd/update-collection-v3/Makefile (hunk 1 above)
Comment (rnishtala-sumo) [reply src/go/cmd/update-collection-v3/Makefile]: added a loop for this
```
build: $(eval SHELL:=/bin/bash)
	for os_arch in linux_amd64 linux_arm64 darwin_arm64 darwin_amd64 windows_arm64 windows_amd64; do \
		os=`echo $$os_arch | awk -F_ '{print $$1}'`; \
		arch=`echo $$os_arch | awk -F_ '{print $$2}'`; \
		env GOOS=$$os GOARCH=$$arch go build -o update-collection-v3-sumo-$$os_arch; \
	done
```

**Comment by rnishtala-sumo** [comment]
After adding a loop in the makefile to support all 6 distributions, the artifacts from a test run is below:
https://github.com/SumoLogic/sumologic-kubernetes-tools/actions/runs/3941030397

**Pull request merged**